saleor/payment/utils.py

```diff
@@ -495,3 +517,28 @@ def gateway_refund(payment, amount: Decimal = None) -> Transaction:
 
     _gateway_postprocess(transaction, payment)
     return transaction
+
+
+def extract_id_for_payment_gateway(user, gateway):
+    key = prepare_label_name(gateway)
```

Comment:
Shouldn't this live inside getters/setters for meta? At this level, I only want to get a value or null and not care about how keys are internally stored.

Reply:
Question is whether we have any requirements for the format of the keys, e.g. that they have to be always stripped and uppercase.

Reply:
I just added this for safety in gateway customer-id related stuff, not really sure if this should apply to all _meta_?

Reply:
Probably discussion should continue in #4425